Rc1 doc updates (#151)
SolarArbiter/SolarArbiter.github.io: documentation/dashboard/working_with_data.md

[API documentation](https://api.solarforecastarbiter.org/)
for details.

### Limits on Download size

Data downloads are limited to a maximum of one year of data per request. Users
may download more than one year of data by making multiple requests.

> 

Author: I'll work on adding it. Just discovered that the flask `MAX_CONTENT_LENGTH` config isn't doing what we want.

> 

Author: handled in https://github.com/SolarArbiter/solarforecastarbiter-api/pull/261

    Click the **Create new Report** link.
    <img class="my-3" src="/images/reports.png"/>

2.  Select the type of report from the _Report Type_ menu at the top of the
    page. The report type dictates the type of forecast that will be analyzed
    in the report. Each forecast type (deterministic, event, and probabilistic)
    will have different available metrics. Event forecasts do not support
    reference forecasts or uncertainty.

3.  Enter the report name and a start and end for the period to analyze.
    Create pairs of Observations and Forecasts by following these steps:
    - Select whether to compare the forecast to an Aggregate or Observation
      using the 'Compare Forecast to' field.
    - If comparing the Forecast to an Observation, select a Site. The
      forecast field will populate with a list of forecasts located at the
      site.

>     - Select a Forecast. The Observation (or Aggregate) and Reference Forecast

Reviewer: ```suggestion
    - Select a Forecast. Use the Variable menu to limit the displayed forecasts to specific variable. The Observation (or Aggregate) and Reference Forecast
```

      fields will be populated with options that match the forecast's site or
      aggregate and variable.
    - Select an Observation or Aggregate.
    - (Optional) Select a Reference Forecast. Specifying a reference forecast
      is only required when calculating the [forecast skill](/metrics/#s)
      metric, and are not applicable for event forecasts.
    - Choose how to define uncertainty. You may choose to ignore uncertainty,

>       use the uncertainty value defined for the observation or provide your

Reviewer: ```suggestion
      use the uncertainty value defined in the observation metadata, or provide your
```

      own value. The default option is to ignore uncertainty. Uncertainty
      cannot be defined for event forecasts.
    - Click the **Add Forecast, Observation pair**.

    Multiple Pairs can be added by repeating this process. Pairs can be removed
    by clicking the red 'remove' link on the right side of the forecast &
    observation box.

>      <img class="my-3" src="/images/report_form.png"/>

Reviewer: the new image here does not have a link to probabilistic forecasts. I know that it's hidden right now but I am going to get the core code working before this is released. I'm also wondering if we need any further explanation for the probabilistic form and where it would go.

    After clicking submit, you will be returned to the report listing page where
    you will see the newly created report with a status of **pending**. The Arbiter
    will process the report and then set its status to **complete** or
    **failed**. You may then view the web version of the report. Failed reports
    will contain an error message about the failure.

> Results
> -------
> 
> The Solar Forecast Arbiter is foremost a forecast evaluation platform. This
> section explains how to interpret the results of those evaluations.
> 
> ### Data Validation Results

Reviewer: ```suggestion
Data Validation
---------------
```

> The Solar Forecast Arbiter performs validation on all Observation data. The

Reviewer: ```suggestion
The Solar Forecast Arbiter performs validation on all Observation data. It does not validate any forecast data. The
```

data validation toolkit applies quality flags to values in observation data.

> The Solar Forecast Arbiter includes these quality flags with observation data

Reviewer: ```suggestion
The Solar Forecast Arbiter includes these quality flags with downloaded observation data
```

> in the form of a bit mask. The table below shows quality flags and their

Reviewer: ```suggestion
in the form of a bit mask. Users can implement their own bit mask parsing functions or use the the [parsing functions available in solarforecastarbiter-core](https://solarforecastarbiter-core.readthedocs.io/en/latest/api.html#quality-flag-mapping). The table below shows quality flags and their
```

> bitmask.

Reviewer: Also need to say something about derived quality flags since they're available in the report form and shown on the observation page, but not available for download. Tricky.

{: .table}

> |Quality Flag|Bitmask|Integer Value|Description|

Reviewer: bitmask column should appear as code. either use css or brute force it with double ticks in each cell.

|------------|-------|-------------|-----------|
|OK|0000000000000000|0|No validation flags.|

> |USER FLAGGED|0000000000000001|1|Providing user flagged data as problematic.|

Reviewer: ```suggestion
|USER FLAGGED|0000000000000001|1|User flagged data as problematic before uploading to Arbiter.|
```

|NIGHTTIME|0000000000010000|16|Value occurs at night.|
|CLEARSKY|0000000000010000|32|Values consistent with clear sky condition.|

> |SHADED|0000000001000000|64|Values indicate module shading.|

Reviewer: ```suggestion
|SHADED|0000000001000000|64|Values indicate module shading. Not implemented.|
```

|UNEVEN FREQUENCY|0000000010000000|128|Difference in timestamp between value and previous value does not conform to observation frequency.|

> |LIMITS EXCEEDED|0000000100000000|256|Physical limits exceeded. Uses QCrad criteria.|

Reviewer: ```suggestion
|LIMITS EXCEEDED|0000000100000000|256|Physical limits exceeded. Uses QCrad criteria for irradiance, DC or AC capacity for power.|
```

|CLEARSKY EXCEEDED|0000010000000000|512|Value greater than clearsky value.|

> |STALE VALUES|0000100000000000|1024||

Reviewer: ```suggestion
|STALE VALUES|0000100000000000|1024|Values are unchanged for several intervals, suggesting an issue with communications.|
```

|INTERPOLATED VALUES|0001000000000000|2048|Values appear linear, suggesting interpolation.|
|CLIPPED VALUES|0010000000000000|4096|Values indicate possible clipped power levels.|
|INCONSISTENT IRRADIANCE COMPONENTS|0100000000000000|8192|GHI, DHI, and DNI are inconsistent.|

> |DAILY VALIDATION APPLIED|1000000000000000|16384||

Reviewer: ```suggestion
|DAILY VALIDATION APPLIED|1000000000000000|16384|The Arbiter has analyzed this point in the context of all of the points around it.|
```

> #### Quality Flags on the Dashboard

Reviewer: ```suggestion
### Quality Flags on the Dashboard
```

On the Dashboard quality flags are plotted along with their timeseries values.
Each type of quality flag has it's own colored bar running parallel to the x
axis of the timeseries plots. A vertical colored bar indicates that the data
was flagged at that timestamp while a solid colored blocks indicate the data
was flagged within the colored range.

> <img class="my-3" src="/images/quality_flag_plot.png"/>

Reviewer: this image looks pixelated

### Report Analysis Results

Solar Forecast Arbiter reports contain timeseries plots as well as a plot for
 each metric and category selected during report creation

> (see [Create New Report](/dashboard/working-with-data/#create-new-report)).

Reviewer: This doesn't do much for me. Maybe it would be worthwhile to add a report documentation section but I think they're self explanatory enough to leave for a later version.
